test/ComponentInstallerTest.php

```diff
@@ -588,52 +613,48 @@ public function getModuleDependencies()
+        $package->method('getExtra')->willReturn([
             'laminas' => [
                 'component' => $packageName,
             ],
         ]);
-        $package->getAutoload()->willReturn([
+        $package->method('getAutoload')->willReturn([
             $autoloading => $autoload,
         ]);
 
-        $this->installationManager->getInstallPath(Argument::exact($package->reveal()))
+        $this->installationManager
+            ->method('getInstallPath')
+            ->with($package)
             ->willReturn(vfsStream::url('project/' . $installPath));
 
-        $operation = $this->prophesize(InstallOperation::class);
-        $operation->getPackage()->willReturn($package->reveal());
+        $operation = $this->createMock(InstallOperation::class);
+        $operation->method('getPackage')->willReturn($package);
 
-        /** @psalm-var PackageEvent&ObjectProphecy $event */
-        $event = $this->prophesize(PackageEvent::class);
-        $event->isDevMode()->willReturn(true);
-        $event->getOperation()->willReturn($operation->reveal());
+        $event = $this->createMock(PackageEvent::class);
+        $event->method('isDevMode')->willReturn(true);
+        $event->method('getOperation')->willReturn($operation);
         $this->prepareEventForPackageProviderDetection($event, 'some/component');
 
-        $this->rootPackage->getRequires()->willReturn([]);
-        $this->rootPackage
-            ->getName()
-            ->willReturn('some/component');
+        $this->rootPackage->method('getName')->willReturn('some/component');
 
-        $this->io->ask(Argument::that(function ($argument) use ($packageName) {
-            return ComponentInstallerTest::assertPrompt($argument, $packageName);
-        }), 1)->willReturn(1);
-
-        $this->io->ask(Argument::that(function ($argument) {
-            return ComponentInstallerTest::assertPrompt($argument);
-        }), 'y')->willReturn('y');
-
-        $this->io->write(Argument::that(function ($argument) use ($packageName) {
-            return strstr($argument, sprintf('Installing %s from package some/component', $packageName));
-        }))->shouldBeCalled();
+        $this->createOutputAssertions([
+            sprintf('Installing %s from package some/component', $packageName),
+        ]);
+        $this->createInputAssertions([
+            RememberedAnswerQuestionAssertion::inject($packageName, 1, true),
+        ]);
 
-        $this->assertNull($this->installer->onPostPackageInstall($event->reveal()));
+        $this->installer->onPostPackageInstall($event);
 
-        $config = include vfsStream::url('project/config/application.config.php');
+        /** @psalm-suppress UnresolvableInclude */
+        $config = require vfsStream::url('project/config/application.config.php');
+        Assert::isNonEmptyMap($config);
+        Assert::keyExists($config, 'modules');
+        Assert::isNonEmptyList($config['modules']);
```

Comment:
If you're in a test case... why are you using webmozart/assert, instead of PHPUnit assertions?

I think these are all valid things to assert; I just think that, in terms of usability and discoverability of issues, having these as PHPUnit assertions will make it easier to understand when something changed, and where; in particular, when we consider running in GHA, we have reporters that create annotations that help the patch author identify where the failures are, without needing to dig through logs...

Reply:
As far as I know, phpunit assertions do not provide `@psalm-assert` annotations and thus do not provide proper types to psalm.
On top of this, there is neither a phpunit assertion to verify if an array is a map or a list.

I've changed this to `@psalm-var` annotations instead.